e2e/homepage.spec.ts

```diff
@@ -1,11 +1,19 @@
+import { testWithII } from '@dfinity/internet-identity-playwright';
 import { expect, test } from '@playwright/test';
+import { Homepage, HomepageLoggedIn } from './utils/pages/homepage.page';
 
-const testUrl = '/';
+test('should display homepage in logged out state', async ({ page }) => {
+	const homepage = new Homepage(page);
 
-test('should display not logged in homepage', async ({ page }) => {
-	await page.goto(testUrl);
+	await homepage.waitForLoggedOut();
 
-	await page.getByTestId('login-button').waitFor();
+	await expect(homepage.page).toHaveScreenshot({ fullPage: true });
```

Comment:
I would not leak the `page`, feels a bit like an anti pattern but, that's maybe an opiniated answer. I would use the page available in the function or defer the screenshot with the page object.

Reply:
Using the original `page` to take a screenshot sounds clean to me. Updated.